### PR TITLE
WIP: when switching to near me mode, sort by distance.

### PR DESCRIPTION
**Pull request by FlipperPA**
Howdy @joshreisner - I'm taking a stab at sorting by distance when a user switches to Near Me mode. I've hit an impasse, and Google hasn't been much help. Every time I think I'm getting better at ReactJS, I step away a few weeks, and its back to square one! :)

Here's a WIP branch - what I'm expecting is that the `console.log` statements should have the arrays in different orders. The first should have the initial order (by time), and the second should be sorted by the distance. However, the second is still sorted by the distance. Any idea why?

You can see the diff, but here are the console logs in question:

https://github.com/code4recovery/react/blob/feature/nearme-sort-distance/src/helpers/data.js#L62

I'll keep pounding on it, but I'm wondering if you see something obvious.

### CODE REVIEW

**Comment by FlipperPA** [comment]
This example may shed some light, but I'm not understanding why.

```javascript
let x = []

x = [
    {"name": "erika", "distance": 0.123},
    {"name": "andrew", "distance": 5.1234},
    {"name": "david", "distance": 25.1234},
    {"name": "chris", "distance": 3.21},
    {"name": "bart", "distance": 4.321}
];

x.sort((a, b) => a.name > b.name);
console.log(x);

x.sort((a, b) => a.distance - b.distance);
console.log(x);
```

I'd expect this to show the list sorted by names first, then a second time sorted by distance. Instead, it shows it twice sorted by distance. If I exclude the second sort and `console.log`, it shows it sorted by name.

**Comment by joshreisner** [comment]
Hey @FlipperPA I think it has to do with objects and references. I believe that `x` in your example is a reference to an array, not a value like a string or number. (Also for this reason I think you could use `const`?)

```
const x = [
  { name: 'erika', distance: 0.123 },
  { name: 'andrew', distance: 5.1234 },
  { name: 'david', distance: 25.1234 },
  { name: 'chris', distance: 3.21 },
  { name: 'bart', distance: 4.321 },
];

const y = [...x];

x.sort((a, b) => a.name.localeCompare(b.name));
console.log(x);

y.sort((a, b) => a.distance - b.distance);
console.log(y);
```

All this said, we don't actually sort meetings by distance in the plugin. We filter by distance, so that users may find meetings within, say, 2 miles, but meetings are always sorted by time.

File: src/helpers/distance.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export function distanceNumber(a, b) {
 }
 
 export function distance(a) {
-  if(a) {
+  if (a) {
```
Comment (joshreisner) [inline src/helpers/distance.js]: If you install Prettier and set to format on save it could take care of this stuff for you automatically.

**Comment by joshreisner** [comment]
Looks like we already have a ticket for the distance filtering: https://github.com/code4recovery/react/issues/37

**Comment by FlipperPA** [comment]
Thanks @joshreisner - my love to JavaScript is not increasing. :)

I was looking for a quick improvement while waiting for a friend to work on the distance filter, but I'm guessing time might be better spent just doing it.

**Pull request closed without merging**